x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/TimingStatsReporter.java

```diff
@@ -35,8 +36,9 @@ public TimingStats getCurrentTimingStats() {
         return new TimingStats(currentTimingStats);
     }
 
-    public void reportBucketProcessingTime(long bucketProcessingTimeMs) {
-        currentTimingStats.updateStats(bucketProcessingTimeMs);
+    public void reportBucket(Bucket bucket) {
+        currentTimingStats.updateStats(bucket.getProcessingTimeMs());
+        currentTimingStats.setLatestRecordTimestamp(bucket.getTimestamp().toInstant());
```

Comment:
Bucket timestamp is the start of the bucket. If the timestamp falls into the next window for calculating exp average then that new window should have the incremented value. With the datafeed the lastest record timestamp represents the end of the bucket and should count in the previous.

In this case the the method you should be calling is set*Earliest*RecordTimesamp